GW2EIEvtcParser/EncounterLogic/Raids/W7/PeerlessQadim.cs

```diff
@@ -335,25 +384,195 @@ internal override void ComputePlayerCombatReplayActors(AbstractPlayer p, ParsedE
             var magmaDrop = p.GetBuffStatus(log, MagmaDrop, log.FightData.FightStart, log.FightData.FightEnd).Where(x => x.Value > 0).ToList();
             int magmaRadius = 420;
             int magmaOffset = 4000;
+            int magmaDuration = 600000;
             string[] magmaColors = { "255, 215, 0", "255, 130, 50" };
             int magmaColor = 0;
             foreach (Segment seg in magmaDrop)
             {
-                int magmaDropEnd = (int)seg.End;
+                // If a player has gone into downstate while in air, the magma field will not spawn
+                var hasPlayerDownedInAir = p.IsDowned(log, seg.Start, seg.End + 350);
```

Comment:
Ok I see, that is not the correct API to use, isDowned returns true if actor is currently in downed state, if you are looking for the moment when downstate happens you should use CombatData.GetDownedEvents